src/wp-includes/class-wp-http.php

```diff
@@ -467,9 +467,9 @@ static function ( $attr ) {
 						return null !== $attr;
 					}
 				);
-				$cookie_jar[ $value->name ] = new WpOrg\Requests\Cookie( $value->name, $value->value, $attributes, array( 'host-only' => $value->host_only ) );
+				$cookie_jar[ $value->name ] = new WpOrg\Requests\Cookie( (string) $value->name, $value->value, $attributes, array( 'host-only' => $value->host_only ) );
```

Comment:
Strictly speaking, `Wp_Http_Cookie` should always have a string for the name already, but there is no harm to being specific here. 👍🏼